addons/account/views/account_move_views.xml

```diff
@@ -332,7 +332,11 @@
                     <separator/>
                     <filter string="To check" name="to_check" domain="[('move_id.checked', '=', False), ('parent_state', '!=', 'draft')]"/>
                     <separator/>
-                    <filter string="Unreconciled" domain="[('account_id.reconcile', '=', True)]" name="reconcilable_account" help="Journal items where the account allows reconciliation no matter the residual amount"/>
+                    <filter string="Unreconciled"
+                            domain="[('account_id.reconcile', '=', True), '|', ('matching_number', '=', False), ('full_reconcile_id', '=', False)]"
```

Comment:
Why do you need to check the `full_reconcile_id` ? I think  the check on matching_number is enough.